pkg/domain/runaway.go

```diff
@@ -140,20 +141,49 @@ func (do *Domain) deleteExpiredRows(tableName, colName string, expiredDuration t
 	}
 }
 
+func (do *Domain) runawayStartLoop() {
+	defer util.Recover(metrics.LabelDomain, "runawayStartLoop", nil, false)
+	runawayWatchSyncTicker := time.NewTicker(runawayWatchSyncInterval)
+	count := 0
+	var err error
+	logutil.BgLogger().Info(
+		"try to start runaway manager loop",
+		zap.Error(err))
+	for {
+		select {
+		case <-do.exit:
+			return
+		case <-runawayWatchSyncTicker.C:
+			// Due to the watch and watch done tables is created later than runaway queries table
+			err = do.updateNewAndDoneWatch()
+			if err == nil {
+				do.wg.Run(do.runawayRecordFlushLoop, "runawayRecordFlushLoop")
+				do.wg.Run(do.runawayWatchSyncLoop, "runawayWatchSyncLoop")
+				do.runawayManager.MarkSyncerInitialized()
+				return
+			}
+		}
+		count++
+		if count == 1800 {
```

Comment:
every 1800 seconds?

Reply:
Yes, I think we should log it every appropriate time interval